Automattic/wp-calypso
# Ensure loginSocialUser is called to create a session

Related to 7559-gh-Automattic/dotcom-forge

## Proposed Changes

* When the user is not logged in visits `/start` and logs in via social login like: Google, Github, or Apple the app state is sill logged out until the start flow finishes.
* This can lead to potential issue if the start flow needs to redirect to another place
* This can be easily reproduced if you open WordPress.com in a different tab. The logged out landing page will be shown even though the user has logged in using social login.

## Why are these changes being made?


* In some cases flows living in `start` might need to redirect to another flow which depends on the user logged in state.

## Testing Instructions



* Follow the instruction from p4TIVU-9su-p2 and apply the PoC changes from https://github.com/Automattic/wp-calypso/pull/91208 to this PR **OR** use [this branch](https://github.com/Automattic/wp-calypso/tree/tmp/user-step-social-login-test) 
* Visit `https://wordpress.com/start`
* Test the login with Google, Apple and Github. After login open a new tab for WordPress.com and you should see the logged in view.


## Pre-merge Checklist
